metricbeat-tests/runner_test.go

```diff
@@ -54,78 +58,85 @@ func TestMain(m *testing.M) {
 	os.Exit(status)
 }
 
-func cleanUpOutputs() {
-	dir, _ := os.Getwd()
-
-	pattern := dir + "/outputs/*.metrics*"
-
-	files, err := filepath.Glob(pattern)
-	if err != nil {
-		log.WithFields(log.Fields{
-			"error": err,
-			"path":  pattern,
-		}).Fatal("Cannot retrieve outputs :(")
+// assertHitsArePresent returns an error if no hits are present
+func assertHitsArePresent(hits map[string]interface{}, q ElasticsearchQuery) error {
+	hitsCount := len(hits["hits"].(map[string]interface{})["hits"].([]interface{}))
```

Comment:
Each one of these type conversions can panic if they don't exist or are not of the expected type, you may want to convert item by item:
```suggestion
      var hits []interface{}
      if hitsObject, ok := hits["hits"].(map[string]interface{}); ok {
          hits, _ = hitsObject["hits"].([]interface{})
      }
      hitsCount = len(hints)
```

Or use something like [mapstructure](https://godoc.org/github.com/mitchellh/mapstructure):
```suggestion
      var searchResponse struct {
          Hints struct {
              Hints []interface{}
          }
      }
      mapstructure.Decode(hits, &searchResponse)
      hitsCount := len(searchResponse.Hints.Hints)
```

Reply:
The idea is to replace this `map[string]interface{}` with a fluent approach to create queries. I used this as in the Elasticsearch Go SDK